Configuration/Services.php

```diff
@@ -58,5 +58,5 @@ static function (
         ->setFactory([new Reference(TreeMapperFactory::class), 'create'])
         ->setAutowired(true)
         ->setAutoconfigured(true)
-        ->setPublic(false);
+        ->setPublic(true);
```

Comment:
I'd keep `TreeMapper` private. That one is a classic "service" that should always be injected.

Reply:
The tests would agree.